beacon-chain/powchain/log_processing.go

```diff
@@ -35,6 +35,10 @@ const maxTolerableDifference = 50
 const defaultEth1HeaderReqLimit = uint64(1000)
 const depositlogRequestLimit = 10000
 
+func tooMuchDataRequestedError(err error) bool {
+	return err.Error() == "query returned more than 10000 results"
```

Comment:
I don't like that we are basically hard-coding it here to infura only. But given that the external providers don't have any standardized error messages, this will have to do for now. Can you add a comment that this error is only infura specific.(other providers might have different error messages)